tests/drivers/can/api/src/classic.c

```diff
@@ -587,6 +740,41 @@ ZTEST_USER(can_classic, test_send_and_forget)
 	send_test_frame(can_dev, &test_std_frame_1);
 }
 
+/**
+ * @brief Test error when sending invalid frame (too big data payload).
+ *
+ * This basic test work since the CAN controller is in loopback mode and
+ * therefore ACKs its own frame.
+ */
+ZTEST_USER(can_classic, test_send_invalid_frame)
+{
+	struct can_frame test_invalid_frame = {
+		.flags   = 0,
+		.id      = TEST_CAN_STD_ID_1,
+		.dlc     = 9,
+		.data    = {0}
+	};
+	int err;
+
+	err = can_send(can_dev, (const struct can_frame *) &test_invalid_frame, TEST_SEND_TIMEOUT, NULL, NULL);
+	zassert_not_equal(err, -EBUSY, "arbitration lost in loopback mode");
+	zassert_equal(err, -EINVAL, "send incorrect frame (err %d)", err);
+
+	test_invalid_frame.flags = CAN_FRAME_IDE;
+	test_invalid_frame.dlc = 65;
+
+	err = can_send(can_dev, (const struct can_frame *) &test_invalid_frame, TEST_SEND_TIMEOUT, NULL, NULL);
+	zassert_not_equal(err, -EBUSY, "arbitration lost in loopback mode");
+	zassert_equal(err, -EINVAL, "send incorrect frame (err %d)", err);
+
+	test_invalid_frame.flags = 	CAN_FRAME_ESI;
+	test_invalid_frame.dlc = 4;
+
+	err = can_send(can_dev, (const struct can_frame *) &test_invalid_frame, TEST_SEND_TIMEOUT, NULL, NULL);
+	zassert_not_equal(err, -EBUSY, "arbitration lost in loopback mode");
+	zassert_equal(err, -ENOTSUP, "send incorrect frame (err %d)", err);
+}
```

Comment:
?

Reply:
Please provide more details.

It looks like CAN_FRAME_ESI can NOT be send.
Nor in FD Mode
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcan.c#L936
Nor in Normal Mode
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/can/can_mcan.c#L948
However, this is expected "Only valid in combination with CAN_FRAME_FDF"
https://docs.zephyrproject.org/latest/doxygen/html/group__can__interface.html#ga83f8b7af6eb45e43aaac355de3e69e52

I can move it to a separate test...